model/resolver/WDQSResource.php

```diff
@@ -6,7 +6,6 @@ class WDQSResource extends RemoteResource
     const WDQS_ENDPOINT = "https://query.wikidata.org/sparql";
 
     public function resolve(int $timeout) : ?EasyRdf\Resource {
-        $client = new EasyRdf\Sparql\Client(self::WDQS_ENDPOINT);
```

Comment:
Re-declared in the code below

Reply:
Good catch!

Now that I look at it, it is a bit questionable why lines https://github.com/NatLibFi/Skosmos/blob/4774577479d618932ba68c44c2e59ff1af050c9d/model/resolver/WDQSResource.php#L10-L44 are in the `try` block, though. I would expect them to succeed in any case.

Reply:
Hmm, good point. We can move it out maybe in another PR? One thing that really bothers me in the code is the use of `return $this->showError…` functions, while these `showError`-etc functions actually do not return anything; but instead set HTTP `header()` values, print to the stdout with `echo()` and need the stream to be closed (with `return` or `exit`) somewhere.

e.g.

- https://github.com/NatLibFi/Skosmos/blob/c84dc970cea2954c76c98254c20f72d6da33756a/controller/RestController.php#L49 returns the value of https://github.com/NatLibFi/Skosmos/blob/c84dc970cea2954c76c98254c20f72d6da33756a/controller/Controller.php#L179-L184 (which doesn't actually return anything).

Reply:
Yes, best to leave it to another PR, this one is rock solid as it is! 😃 

Heh, now that you pointed at it, it surely looks ugly. I also remember having some trouble with those things whilst trying to silence the extra output PHPUnit was printing whilst executing tests.

+1 for another PR.